pkg/resources/grant_ownership.go

```diff
@@ -0,0 +1,668 @@
+		sdk.ObjectTypeDynamicTable,
+		sdk.ObjectTypeEventTable,
+		sdk.ObjectTypeExternalTable,
+		sdk.ObjectTypeFileFormat,
+		sdk.ObjectTypeFunction,
+		sdk.ObjectTypeHybridTable,
+		sdk.ObjectTypeIcebergTable,
+		sdk.ObjectTypeImageRepository,
+		sdk.ObjectTypeMaterializedView,
+		sdk.ObjectTypeNetworkRule,
+		sdk.ObjectTypePackagesPolicy,
+		sdk.ObjectTypePipe,
+		sdk.ObjectTypeProcedure,
+		sdk.ObjectTypeMaskingPolicy,
+		sdk.ObjectTypePasswordPolicy,
+		sdk.ObjectTypeProjectionPolicy,
+		sdk.ObjectTypeRowAccessPolicy,
+		sdk.ObjectTypeSessionPolicy,
+		sdk.ObjectTypeSecret,
+		sdk.ObjectTypeSequence,
+		sdk.ObjectTypeStage,
+		sdk.ObjectTypeStream,
+		sdk.ObjectTypeTable,
+		sdk.ObjectTypeTag,
+		sdk.ObjectTypeTask,
+		sdk.ObjectTypeView:
+		if _, ok := identifier.(sdk.SchemaObjectIdentifier); !ok {
+			return nil, sdk.NewError(fmt.Sprintf("invalid object_name %s, expected schema object identifier", objectName))
+		}
+	default:
+		return nil, sdk.NewError(fmt.Sprintf("object_type %s is not supported, please create a feature request for the provider if given object_type should be supported", objectType))
+	}
+
+	return identifier, nil
+}
+
+func getOwnershipGrantOn(d *schema.ResourceData) (sdk.OwnershipGrantOn, error) {
+	var ownershipGrantOn sdk.OwnershipGrantOn
+
+	on := d.Get("on").([]any)[0].(map[string]any)
+	onObjectType := on["object_type"].(string)
+	onObjectName := on["object_name"].(string)
+	onAll := on["all"].([]any)
+	onFuture := on["future"].([]any)
+
+	switch {
+	case len(onObjectType) > 0 && len(onObjectName) > 0:
+		objectType := sdk.ObjectType(strings.ToUpper(onObjectType))
+		objectName, err := getOnObjectIdentifier(objectType, onObjectName)
+		if err != nil {
+			return ownershipGrantOn, err
+		}
+		ownershipGrantOn.Object = &sdk.Object{
+			ObjectType: objectType,
+			Name:       objectName,
+		}
+	case len(onAll) > 0:
+		ownershipGrantOn.All = getGrantOnSchemaObjectIn(onAll[0].(map[string]any))
+	case len(onFuture) > 0:
+		ownershipGrantOn.Future = getGrantOnSchemaObjectIn(onFuture[0].(map[string]any))
+	}
+
+	return ownershipGrantOn, nil
+}
+
+func getOwnershipGrantTo(d *schema.ResourceData) sdk.OwnershipGrantTo {
+	var ownershipGrantTo sdk.OwnershipGrantTo
+
+	if accountRoleName, ok := d.GetOk("account_role_name"); ok {
+		ownershipGrantTo.AccountRoleName = sdk.Pointer(sdk.NewAccountObjectIdentifierFromFullyQualifiedName(accountRoleName.(string)))
+	}
+
+	if databaseRoleName, ok := d.GetOk("database_role_name"); ok {
+		ownershipGrantTo.DatabaseRoleName = sdk.Pointer(sdk.NewDatabaseObjectIdentifierFromFullyQualifiedName(databaseRoleName.(string)))
+	}
+
+	return ownershipGrantTo
+}
+
+func getOwnershipGrantOpts(id *GrantOwnershipId) *sdk.GrantOwnershipOptions {
+	if id != nil && id.OutboundPrivilegesBehavior != nil {
+		outboundPrivileges := id.OutboundPrivilegesBehavior.ToOwnershipCurrentGrantsOutboundPrivileges()
+		if outboundPrivileges != nil {
+			return &sdk.GrantOwnershipOptions{
```

Comment:
it would be easier to have just one return in this method, so:
1. create options at the start
2. set outbound privileges inside if
3. return at the end